src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -255,5 +255,31 @@ protected virtual Task MapAsync(HttpContext httpContext, Route route, List<HttpC
 
         var aggregator = _factory.Get(route);
         return aggregator.Aggregate(route, httpContext, contexts);
+    }
+
+    protected virtual async Task<Stream> CloneRequestBodyAsync(HttpRequest request, CancellationToken aborted)
+    {
+        request.EnableBuffering();
+        if (request.Body.Position == 0)

```

Comment:
why not try to reduce nesting?
```c#
if (request.Body.Position != 0)
{
    return request.Body;
}
```

Reply:
Make sense! 😄

Reply:
Fixed by https://github.com/ThreeMammals/Ocelot/pull/2050/commits/33d1743311d1e8c4bef27a656101ad5537fc11af

Reply:
I'm not a fan of deep nesting because the code can become very difficult to read. If we agree on a specific pattern we should follow it and we tend to use this pattern in later implementations in Ocelot.